saphir/src/redirect.rs

```diff
@@ -71,6 +73,12 @@ impl Builder {
         self
     }
 
+    #[inline]
+    pub fn add_header<K: AsRef<str>, V: AsRef<str>>(mut self, key: K, value: V) -> Self {
```

Comment:
this function should be named `header` and the signature should be :
```Rust 
#[inline]
    pub fn header<K, V>(mut self, key: K, value: V) -> Self
    where
        HeaderName: TryFrom<K>,
        <HeaderName as TryFrom<K>>::Error: Into<http::Error>,
        HeaderValue: TryFrom<V>,
        <HeaderValue as TryFrom<V>>::Error: Into<http::Error>,
    {  ...  }
```
This will be consistent with the `header` fn on the `response::Builder` type

Reply:
Agreed for the signature,. The only concern I have is that the existing api alter the associated item instead of adding to it (fragment_param replace the whole param part instead of appending a param, query_param does the same). the add_ prefix help indicate this difference in behavior.

Reply:
it's based on what is set, in the case add ins't clearer since it doesn't not respect consistancy with the response Builder type, fir other function like say: `status` we don't need anything different since only one status in possible, the same goes for query or fragment, you can't have multiple frangment, so you set it once

Reply:
Okay, it does fit well with the current interface.
I would say that while fragment is indeed a single value, in which we can represent a collection in wathever string format we want, a query string is a set format for key-value, not an arbitrary string following '?'. In that regard it's kind of weird to consider it as a single opaque object, but, making it behave like fragment, which should not impose a format, feels like an acceptable rationale.

Will be changed to `header()`.